libs/vscode/nx-project-view/src/lib/helper.ts

```diff
@@ -0,0 +1,48 @@
+import { ProjectConfiguration } from "@nrwl/devkit";
+
+export class NxProjectTreeHelper {
+
+  static async getProjectsInFolders(projects: [string, ProjectConfiguration][]): Promise<Record<string,any>> {
+    const projectsObject = {};
+
+    projects.forEach((project) => {
+      const lastArray = project[1] as ProjectConfiguration;
+      const splittedProject = lastArray.root.split('/');
+
+      splittedProject.reduce((r: {[key: string]: string;} & any, e: string, i: number) => {
+          if (splittedProject.length -1 > i) {
+              return r[e] || (r[e] = {});
+          }
+          return r[e] || (r[e] = project)
+      }, projectsObject)
```

Comment:
> This was a bit too confusing because of the variable names, and the modifying nested properties of the projectObject property. I feel like this (or something close to it) would make this a little more readable:

I've tried implementing your code snippet, but it doesn't compile, so i've added functional naming for the parameters.

Reply:
the variable names were one part of the confusion, but it's still modifying a nested object. There's a bunch of redirection happening with the nested object